File: euler010-proc.php
#!/usr/bin/env php
<?php
/** Problem 10: Summation of primes
 * https://projecteuler.net/problem=10
 */

function euler010(int $bound): int
{
    $primes = [2];
    for ($candidate = 3; $candidate < $bound; $candidate += 2) {
        foreach($primes as $prime) {
            if ($prime ** 2 > $candidate) {
                break;
            }
            if ($candidate % $prime == 0) {
                continue 2;
            }
        }
        $primes[] = $candidate;
    }
    return array_sum($primes);
}

// Main
echo euler010(intval(trim(file_get_contents('php://STDIN')))) . PHP_EOL;

File: euler010-proc.py
#!/usr/bin/env python3

# Problem 10: Summation of primes
# https://projecteuler.net/problem=10

import sys

def is_prime(primes, number):
    for prime in primes:
        if number % prime == 0:
            return False
        if prime ** 2 > number:
            break
    return True

def euler010(bound):
    primes = [2]
    for candidate in range(3, bound, 2):
        if is_prime(primes, candidate):
            primes.append(candidate)
    return sum(primes)

def parse_input(lines):
    return int(lines[0].strip())

if __name__ == "__main__":
    print(euler010(parse_input(sys.stdin.readlines())))
